pgsql-flexible-server.tf

```diff
@@ -72,3 +97,57 @@ resource "azurerm_postgresql_flexible_server_configuration" "pgsql_server_config
   server_id = azurerm_postgresql_flexible_server.pgsql_server.id
   value     = each.value.value
 }
+
+
+resource "azurerm_postgresql_flexible_server_active_directory_administrator" "pgsql_adadmin" {
+  server_name         = azurerm_postgresql_flexible_server.pgsql_server.name
+  resource_group_name = azurerm_postgresql_flexible_server.pgsql_server.resource_group_name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  object_id           = data.azuread_group.db_admin.object_id
+  principal_name      = local.admin_group
+  principal_type      = "Group"
+}
+resource "azurerm_postgresql_flexible_server_active_directory_administrator" "pgsql_jenkins_admin" {
+  server_name         = azurerm_postgresql_flexible_server.pgsql_server.name
+  resource_group_name = azurerm_postgresql_flexible_server.pgsql_server.resource_group_name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  object_id           = data.azuread_group.mi_db_admin.object_id
+  principal_name      = local.mi_name
+  principal_type      = "ServicePrincipal"
+}
+# module "aad_role" {
+
+#   source               = "git@github.com:hmcts/terraform-postgresql-aad-role.git?ref=master"
+#   name                 = local.db_reader_user
+#   password             = random_password.password.result
+#   db_name              = azurerm_postgresql_flexible_server_database.pg_databases[0].name
+#   server_name          = local.server_name
+#   pgsql_admin_username = azurerm_postgresql_flexible_server.pgsql_server.administrator_login
+#   pgsql_admin_password = azurerm_postgresql_flexible_server.pgsql_server.administrator_password
+# }
+
+resource "null_resource" "set-user-permissions-additionaldbs" {
+  for_each = {
+    for index, db in var.pgsql_databases :
+    db.name => db
+  }
+  triggers = {
+    script_hash    = filesha256("${path.module}/set-postgres-permissions.bash")
+    name           = local.name
+    db_reader_user = local.db_reader_user
+  }
+
+  provisioner "local-exec" {
+    command = "chmod +x ${path.module}/set-postgres-permissions.bash; ${path.module}/set-postgres-permissions.bash"
```

Comment:
run chmod +x locally on this file, permissions will be preserved if you set it up properly

Reply:
Sure - updated it now